src/json/from_json.cr

```diff
@@ -248,7 +248,7 @@ def Enum.new(pull : JSON::PullParser)
   when .int?
     from_value(pull.read_int)
   when .string?
-    parse(pull.read_string)
+    parse(pull.read_string.camelcase)
```

Comment:
`camelcase` conversion is unnecessary here (dirtto for YAML). `Enum.parse` handles both underscore and camelcase. To make sure, you could add a spec.